src/redux/books/books.js

```diff
@@ -1,47 +1,85 @@
-import { v4 as uuidv4 } from 'uuid';
-
 // Actions
+const GET_BOOKS = 'bookstore-react/books/GET_BOOKS';
 const ADD = 'bookstore-react/books/ADD';
 const REMOVE = 'bookstore-react/books/REMOVE';
 
+// key = kcM4ut7Kvp9QgSxmujh0
+const API_KEY = 'kcM4ut7Kvp9QgSxmujh0';
+
 // Reducer
-export default function booksReducer(
-  state = [
-    {
-      title: 'First Book',
-      author: 'First Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Second Book',
-      author: 'Second Author',
-      id: uuidv4(),
-    },
-    {
-      title: 'Third Book',
-      author: 'Third Author',
-      id: uuidv4(),
-    },
-  ],
-  action = {}
-) {
+export default function booksReducer(state = [], action = {}) {
   switch (action.type) {
+    case GET_BOOKS:
+      return action.payload;
     case ADD:
       return [...state, action.book];
     case REMOVE:
-      return state.filter((book) => book.id !== action.book.id);
+      return state.filter((book) => book.item_id !== action.item_id);
     default:
       return state;
   }
 }
 
 // Action Creators
-export const addBook = (book) => ({
-  type: ADD,
-  book,
-});
+export const getBooks = () => async (dispatch) => {
+  const booksFetch = await fetch(
+    `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${API_KEY}/books`,
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    },
+  ).then((data) => data.json());
+  const books = [];
+  Object.keys(booksFetch).forEach((e) => {
+    books.push({ ...booksFetch[e][0], item_id: e });
+  });
+  dispatch({
+    type: GET_BOOKS,
+    payload: books,
+  });
+};
 
-export const removeBook = (book) => ({
-  type: REMOVE,
-  book,
-});
+export const addBook = (book) => async (dispatch) => {
+  await fetch(
+    `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${API_KEY}/books`,
+    {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        item_id: book.item_id,
+        title: book.title,
+        author: book.author,
+        category: book.category,
+      }),
+    },
+  ).then(
+    dispatch({
+      type: ADD,
+      book,
+    }),
+  );
+};
```

Comment:
- [ ] Kindly note that it is a requirement that you make use of [createasyncthunk](https://redux-toolkit.js.org/api/createAsyncThunk) when making your API calls according to the project requirements.